src/main/src/main/java/org/geoserver/catalog/impl/LayerInfoImpl.java

```diff
@@ -184,13 +184,16 @@ public Set<StyleInfo> getStyles() {
             Set<StyleInfo> remoteStyles = ((WMSLayerInfo) getResource()).getStyles();
             // will be null if remote capability document
             // does not have any Style tags
-            if (remoteStyles != null) return remoteStyles;
-            else if (LOGGER.isLoggable(Level.FINE))
+            if (remoteStyles != null) {
+                styles.remove(this.defaultStyle);
+                return remoteStyles;
+            } else if (LOGGER.isLoggable(Level.FINE)) {
                 LOGGER.fine(
                         "No Default Styles found on cascaded WMS Resource"
                                 + getResource().getName());
+            }
         }
-
+        if (styles != null) styles.remove(this.defaultStyle);
```

Comment:
@jodygarnett what do you think of this change?

On one side, it's the most confined change to solve the problem, and auto-fixes a situation that seems just broken. On the other hand, it's changing the configuration.

Maybe compromise, and return a copy of the styles structure, that has the potential default style duplication removed? Just thinking out loud.

Reply:
This "conflict" between the default style and the alternate style list is just a WMS protocol concern - so yes we should make a specific method to make this easier for WMS protocol - and leave the configuration as is.

* I know I have personally configured a lot of servers with several alternate styles, and then completely independently adjusted the default style. It would be frustrating if when trying different default styles the alternate styles were modified. :)
* It is only really the WMS protocol that has this limitation, the ogcapi-api situation may be different.

Can you confirm that the idea of a default style is just a GeoServer configuration idea, the protocol just has an empty ``style=`` and we wanted a way to indicate what was to be used...

Let me check:

> 7.2.4.6.5 Style
> 
> Zero or more Styles may be advertised for a Layer or collection of layers using <Style> elements, each of which shall have <Name> and <Title> elements. The style's Name is used in the Map request STYLES parameter. The Title is a human-readable string. If only a single style is available, that style is known as the “default” style and need not be advertised by the server.

So the idea of a default style is in the spec, and indicates when we could skip listing the style.

Reply:
Andrea I like your proposed approach, it could be a utility method, or added to LayerInfo interface as a default method:

```
default Set<StyleInfo> toStyles(){
    HashSet<StyleInfo> styles = new HashSet<>();
    if (getStyles() != null) styles.addAll(getStyles());
    if (getDefaultStyle() != null) styles.remove(getDefaultStyle());
    return styles;
}
```

Reply:
The PR looks to just cover external styles from WMSLayerInfo; is it worth fixing this for all layers?

Reply:
@jodygarnett if we want to follow pattern with the other interfaces, it should be called just "styles()", all other interfaces remove the getter prefix to indicate that the method returns derivative information (e.g., ``FeatureTypeInfo.attributes()`` vs ``FeatureTypeInfo.getAttributes()``).

Reply:
I agree `styles()` would be appropriate.